### PR TITLE
Adds script to validate sts role arn access

### PR DESCRIPTION
**Pull request by iamkirkbater**
This PR adds a script to validate that your STS role arns are set up correctly by running through the assume-role steps.

### CODE REVIEW

**Comment by drpaneas** [comment]
Nice! Before merging we should add binary validation and also make sure the cluster is up and running as well, along with some additional tests (e.g. for namespace, secrets if they exist, etc) and print appropriate messages.

**Comment by dastergon** [comment]
Can we also incorporate this into the Makefile so we don't have to remember to execute this script? This could go as a sub-target to the `predeploy` target

**Comment by dkeohane** [comment]
Please update the install documentation with this call - to ensure all new installs have their accounts correctly configured

File: Makefile [hunk 1]
```diff
@@ -50,6 +50,16 @@ ifndef OSD_STAGING_1_OU_BASE_ID
 	$(error OSD_STAGING_1_OU_BASE_ID is undefined)
 endif
 
+.PHONY: check-sts-setup
+check-sts-setup: ## Checks if STS roles are set up correctly
+ifndef STS_JUMP_ROLE
+	$(error STS_JUMP_ROLE is undefined. STS_JUMP_ROLE is the ARN of the role we use as a bastion to access the installation role)
```
Comment (dkeohane) [inline Makefile:56]: Could potentially link to our documentation here which explains where they env variables come from:
https://github.com/openshift/aws-account-operator/blob/master/docs/1.1-InstallationPrerequisites.md#1131---jump-role

File: Makefile [hunk 2]
```diff
@@ -260,8 +270,11 @@ predeploy-aws-account-operator: ## Predeploy AWS Account Operator
 	# Create zero size account pool
 	@oc apply -f hack/files/aws.managed.openshift.io_v1alpha1_zero_size_accountpool.yaml
 
+.PHONY: validate-deployment
+validate-deployment: check-aws-account-id-env check-sts-setup ## Validates deployment configuration
+
 .PHONY: predeploy
-predeploy: predeploy-aws-account-operator deploy-aws-account-operator-credentials create-ou-map ## Predeploy Operator
+predeploy: predeploy-aws-account-operator deploy-aws-account-operator-credentials create-ou-map validate-deployment ## Predeploy Operator
```
Comment (dkeohane) [inline Makefile:277]: Should validate-deployment be moved to the beginning of the list as we'll want to validate this info before continuing with the deployment

File: Makefile [hunk 3]
```diff
@@ -357,9 +373,8 @@ check-spell: # Check spelling
 # This *adds* `check-spell` ./hack/scripts/misspell_check.sh the existing `lint` provided by boilerplate
 lint: check-spell
 
-# Test all
 .PHONY: test-all
-test-all: lint clean-operator test test-apis test-account-creation test-ccs test-reuse test-awsfederatedaccountaccess test-awsfederatedrole test-aws-ou-logic test-sts-accountclaim ## Runs all integration tests
+test-all: lint clean-operator test test-apis test-integration ## Runs all tests
```
Comment (dkeohane) [inline Makefile:377]: much cleaner :+1:

File: docs/1.1-InstallationPrerequisites.md [hunk 4]
```diff
@@ -182,6 +182,10 @@ Example minimal permissions Policy Document
 ```
 **Note:** You will need to put the ARN for this `Access Role` into your environment variable, `STS_ROLE_ARN`, in order for this to be used to create accountclaims within the make tests.
 
+### 1.1.3.4 STS Role Validation
```
Comment (dkeohane) [inline docs/1.1-InstallationPrerequisites.md:185]: For visibility could add sample output, but not necessary

**Review by dkeohane — APPROVED** [review]
GJ :100:

**Review by dastergon — COMMENTED** [review]
/lgtm

@dkeohane's suggestions are great to implement before merge

**Pull request merged**